src/httpclient.cc

```diff
@@ -180,6 +180,20 @@ HttpResponse HttpClient::perform(CURL* curl_handler, int retry_times) {
   return response;
 }
 
+HttpResponse HttpClient::download(const std::string& url, curl_write_callback callback, void* userp) {
+  CURL* curl_download = curl_easy_duphandle(curl);
+  curl_easy_setopt(curl_download, CURLOPT_URL, url.c_str());
+  curl_easy_setopt(curl_download, CURLOPT_FOLLOWLOCATION, 0L);
```

Comment:
Is there a reason to deny redirects here?  I think that they are sometimes used by the server to support storing large objects in Amazon S3.